test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,200 @@
+	pod.ObjectMeta.Labels = map[string]string{"test": "liveness-stress"}
+	pod.Spec.Containers[0].LivenessProbe = &livenessProbe
+	pod.Spec.Containers = append(pod.Spec.Containers, make([]v1.Container, numContainers-1)...)
+	for idx := 0; idx < numContainers; idx++ {
+		cc := containerConfig{
+			Name:         "agnhost-container-" + strconv.Itoa(idx+1),
+			ProbeType:    probeType,
+			ContainerIdx: idx,
+		}
+		instance := *pod.Spec.Containers[0].DeepCopy()
+		pod.Spec.Containers[idx] = newContainer(cc, instance)
+	}
+	return pod
+}
+
+func newContainer(config containerConfig, instance v1.Container) v1.Container {
+	newContainerPort := config.ContainerIdx + containerPort + 1
+	instance.Name = config.Name
+	instance.Ports = []v1.ContainerPort{{ContainerPort: int32(newContainerPort)}}
+
+	switch config.ProbeType {
+	case "tcp":
+		instance.LivenessProbe.TCPSocket.Port = intstr.FromInt(newContainerPort)
+		instance.Args = []string{"serve-hostname", "--tcp", "--port", strconv.Itoa(newContainerPort), "--http=false"}
+	case "http":
+		instance.LivenessProbe.HTTPGet.Port = intstr.FromInt(newContainerPort)
+		instance.Args = []string{"serve-hostname", "--http", "--port", strconv.Itoa(newContainerPort)}
+	case "grpc":
+		instance.LivenessProbe.GRPC.Port = int32(newContainerPort)
+		instance.Args = []string{"grpc-health-checking", "--port", strconv.Itoa(newContainerPort), "--http-port", strconv.Itoa(newContainerPort + containerPort)}
+	}
+
+	return instance
+}
+
+func httpGetHandler(path string, port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path,
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func tcpSocketHandler(port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		TCPSocket: &v1.TCPSocketAction{
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func getRestartCount(p *v1.Pod) int {
+	count := 0
+	for _, containerStatus := range p.Status.ContainerStatuses {
+		count += int(containerStatus.RestartCount)
+	}
+	return count
+}
+
+// runLivenessTest verifies the number of restarts.
+func runLivenessTest(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
+	podClient := e2epod.NewPodClient(f)
+	ns := f.Namespace.Name
+	gomega.Expect(pod.Spec.Containers).NotTo(gomega.BeEmpty())
+	// At the end of the test, clean up by removing the pod.
+	ginkgo.DeferCleanup(func(ctx context.Context) error {
+		ginkgo.By("deleting the pod")
+		return podClient.Delete(ctx, pod.Name, *metav1.NewDeleteOptions(0))
+	})
+	ginkgo.By(fmt.Sprintf("Creating pod %s in namespace %s", pod.Name, ns))
+	podClient.Create(ctx, pod)
+
+	// Wait until the pod is running.
+	framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod),
+		fmt.Sprintf("starting pod %s in namespace %s", pod.Name, ns))
+	framework.Logf("Started pod %s in namespace %s", pod.Name, ns)
+
+	deadline := time.Now().Add(defaultObservationTimeout)
+	for start := time.Now(); time.Now().Before(deadline); time.Sleep(10 * time.Second) {
```

Comment:
Would it be more clear if we simply run this loop `defaultObservationTimeout  / 10` times?

Reply:
changed, please check